app/questionnaire/questionnaire_schema.py

```diff
@@ -215,20 +216,52 @@ def get_post_submission(self) -> ImmutableDict:
         schema: ImmutableDict = self.json.get("post_submission", ImmutableDict({}))
         return schema
 
+    def _is_list_name_in_rule(
+        self, when_rules: Union[dict, Sequence], list_name: str
+    ) -> bool:
```

Comment:
I think it is worth dropping the `when`, just `rule` is fine as it can be used outside of a `when`.

Reply:
updated to `rules_dict` , `rules` is being used many times